### PR TITLE
[FAS3] Fix login, register and models

### PR DESCRIPTION
**Pull request by skrzepto**
Resolves issues #195 and #217

### CODE REVIEW

File: fas/models/group.py [hunk 1]
```diff
@@ -248,12 +248,12 @@ class GroupMembership(Base):
     """ A mapping object to SQL GroupMembership table. """
     __tablename__ = 'group_membership'
     id = Column(Integer, primary_key=True)
-    group_id = Column(Integer, ForeignKey('groups.id'))
+    group_id = Column(Integer, ForeignKey('groups.id', ondelete="CASCADE"))  # nullable=False???
```
Comment (pypingou) [inline fas/models/group.py]: The nullable sounds like a good thing, can someone be a member of no group?

File: fas/models/group.py [hunk 2]
```diff
@@ -248,12 +248,12 @@ class GroupMembership(Base):
...
     role = Column(Integer, default=MembershipRole.USER.value)
     status = Column(Integer, default=MembershipStatus.UNAPPROVED.value)
     comment = Column(UnicodeText, nullable=True)
-    person_id = Column(Integer, ForeignKey('people.id'), nullable=False)
-    sponsor_id = Column(Integer, ForeignKey('people.id'), nullable=True)
+    person_id = Column(Integer, ForeignKey('people.id', ondelete="CASCADE"), nullable=False)
+    sponsor_id = Column(Integer, ForeignKey('people.id',  ondelete="SET NULL"), nullable=True)
```
Comment (pypingou) [inline fas/models/group.py]: Isn't there always a sponsor?

File: fas/models/configs.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ class Plugins(Base):
 class AccountPermissions(Base):
     __tablename__ = 'account_permissions'
     id = Column(Integer, primary_key=True)
-    person_id = Column(Integer, ForeignKey('people.id'), nullable=False)
+    person_id = Column(Integer, ForeignKey('people.id', ondelete="CASCADE"), nullable=False)
```
Comment (pypingou) [inline fas/models/configs.py]: While technically correct and probably desirable, I wonder if we really want it, or rather if we want to support deleting users in FAS. From the UI I believe we do not support it, so this is really only something that admins would ran. Kinda still wondering if we even want to allow admins to do it.

@laxathom thoughts?

File: fas/models/group.py (hunk 2 above)
Comment (skrzepto) [inline fas/models/group.py]: @pypingou not sure it was nullable=True before so I figured on delete to have be null. Perhaps changing the sponsor_id to nullable=False would be more desired

File: fas/models/configs.py (hunk 3 above)
Comment (laxathom) [inline fas/models/configs.py]: Yes. We need this feature at a low level. Only admin can delete accounts and groups.
We can also look at providing an option from the config file which disable the feature UI-side.

File: fas/models/group.py (hunk 1 above)
Comment (laxathom) [inline fas/models/group.py]: No, if we have an entry here, this means it is tied up to a group with a status that decribes it.
I'd say can't be nullable.

File: fas/models/group.py (hunk 2 above)
Comment (laxathom) [inline fas/models/group.py]: No, with the new architecture, a group can be live without a sponsor, hence the `nullable=True`

**Comment by skrzepto** [comment]
@laxathom @pypingou updated one of the tables so its nullable=false as stated in the comments.  Any more issues from being considered to be merged?

**Comment by pypingou** [comment]
Looks fine to me

**Comment by laxathom** [comment]
Looks good to me too

**Pull request merged**